assets/javascripts/admin_needle.js

```diff
@@ -1,7 +1,19 @@
 function setupAdminNeedles() {
     function ajaxUrl() {
         var url = $('#needles').data('ajax-url');
-        return url + "?last_match=" + $('#last_match_filter').val() + "&last_seen=" + $('#last_seen_filter').val();
+        var lastMatch;
+        var lastSeen;
+        if ($('#checkbox_custom_last_seen_time').is(':checked')) {
+            lastSeen = getCustomData("last_seen");
+        } else {
+          lastSeen = "last_seen=" + $('#last_seen_filter').val();
+        }
+        if ($('#checkbox_custom_last_match_time').is(':checked')) {
+            lastMatch = getCustomData("last_match");
+        } else {
+            lastMatch = "last_match=" + $('#last_match_filter').val();
+        }
+        return url + "?" + lastMatch + "&" + lastSeen;
```

Comment:
Don't forget to encode these variables properly, e.g. using `encodeURIComponent()`.